### PR TITLE
tests: improve coverage for adwatchd and cmdhandler

### PR DESCRIPTION
**Pull request by GabrielNagy**
UDENG-2764
UDENG-2765

### CODE REVIEW

File: cmd/adwatchd/integration_tests/adwatchd_service_test.go [hunk 1]
```diff
@@ -277,3 +277,12 @@ func TestServiceConfigFlagUsage(t *testing.T) {
...
 }
+
+func TestServiceWithNoNameFails(t *testing.T) {
+	app := commands.New(commands.WithServiceName(""))
+
+	changeAppArgs(t, app, "", "service", "status")
+	err := app.Run()
+
+	require.Error(t, err, "Service with empty name should fail to create")
+}
```
Comment (denisonbarbosa) [inline cmd/adwatchd/integration_tests/adwatchd_service_test.go:288]: I would organize the blocks as:
```suggestion
func TestServiceWithNoNameFails(t *testing.T) {
	app := commands.New(commands.WithServiceName(""))
	changeAppArgs(t, app, "", "service", "status")

	err := app.Run()
	require.Error(t, err, "Service with empty name should fail to create")
}
```
Could be just a biased opinion, but I feel like it makes more sense considering what's happening

File: internal/cmdhandler/cmdhandler_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		wantErr bool
+	}{
+		"Zero args":      {},
+		"Exactly N args": {argCount: 1, args: []string{"arg1"}},
+
+		"Error with less than N args": {argCount: 2, args: []string{"arg1"}, wantErr: true},
+		"Error with more than N args": {argCount: 2, args: []string{"arg1", "arg2", "arg3"}, wantErr: true},
+	}
+
+	for name, tc := range tests {
```
Comment (denisonbarbosa) [inline internal/cmdhandler/cmdhandler_test.go:30]: Nitpick: We usually don't do whitespaces between the end of the map and the loop

**Review by denisonbarbosa — COMMENTED** [review]
Looks good to me, but I do have some small nitpicks :stuck_out_tongue_closed_eyes:

**Review by denisonbarbosa — APPROVED** [review]
Thanks for the changes. Nice work!

**Review by didrocks — APPROVED** [review]
Excellent work!

**Pull request merged**